media/windows/ECC-Ecobee/ECC Pycobee Data.py

```diff
@@ -2326,50 +2410,95 @@ def get_snapshot(conn,
             http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())
-            response = http.request('GET', site_url)
-            logger.debug(F"...checking connection to {site_url}...")
-            logger.debug(F"...response:  {response.headers}")
-            print(F"...checking connection to {site_url}...")
-            print(F"...response:  {response.headers}")
-            return True
-        except Exception as e:
-            logger.debug(F"...connection attempt failed...{e}")
-            print(F"...connection attempt failed...{e}")
+            response = http.request('GET', url)
+            logger.info(F"Internet connection good!")
+            logger.info(F"{response.headers}")
+            print(F"Internet connection good!")
+            print(F"{response.headers}")
+        except (Exception, Error, AttributeError, socket.error) as e:
+            logger.error(F"Error attempting to establish HTTPS connection...{e}")
+            print(F"Error attempting to establish HTTPS connection...{e}")
+            test_success = False
+
+    # Try pinging sites
+    logger.info(F"*** Attempting site pings...")
+    print(F"\n\n*** Attempting site pings...")
+    if 'microsoft.com' in url_root:
+        logger.info(F"...skipping ping test for {url_root}, as site drops ping requests...")
+        print(F"\n...skipping ping test for {url_root}, as site drops ping requests...")
     else:
-        logger.debug(F"Internet connection check retry count exceeded...")
-        print(F"Internet connection check retry count exceeded...")
-        ping_url = 'google.com'
-        logger.debug(F"...now trying ping test to {ping_url}...")
-        print(F"...now trying ping test to {ping_url}...")
-        response_list = ping(ping_url, verbose=True, count=15)
-        logger.debug(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
-        print(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
-        ping_url = 'ecobee.com'
-        print(F"...now trying ping test to {ping_url}...")
-        logger.debug(F"...now trying ping test to {ping_url}...")
-        response_list = ping(ping_url, verbose=True, count=15)
-        logger.debug(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
-        print(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
-
-        return False
+        logger.info(F"Ping test for '{url_root}'...")
+        print(F"\nPing test for '{url_root}'...")
+        try:
+            response_list = ping(url_root, verbose=True, count=15)
+            logger.info(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
+            print(F"Average ping response:  {response_list.rtt_avg_ms}ms.")
+        except (Exception, Error, AttributeError, socket.error) as e:
+            logger.error(F"Error occurred during ping attempt, {e}")
+            print(F"Error occurred during ping attempt, {e}")
+            test_success = False
+
+    # Finally, try doing DNS lookups
```

Comment:
FWIW: If you have already done an HTTP GET and ping, then the DNS lookup for these names has already been performed by the underlying OS.  Hence, this DNS lookup will just return a local value -- it won't actually reach across the internet to resolve the name (because that has already been done).  Also, DNS lookups are cached by the OS, so it's likely that the DNS lookup only happens once an hour, or perhaps even once a day.

Reply:
Sure...if the HTTP GET was successful.  If not, then attempting the DNS lookup should at least show what the lookup returns (or not, if it failed).   I understand that they're cached, but I think displaying the resolved IP(s) and the resolver(s) are useful information if the other tests fail (or if they fail inconsistently). 

Perhaps I should have checked status returns from the HTTP GET and attempted the other tests only if they failed, but not sure it's worth changing.

Reply:
You might want to go from "smallest" to "biggest" test:

1. DNS lookup
1. Ping
1. HTTP/HTTPS

I.e., the DNS test is a prerequisite for the other two to work -- if DNS doesn't work, the other two won't work, either.

Ping is a fundamentally simpler network protocol than HTTP/HTTPS.  I.e., it can verify basic networking connectivity (beyond DNS).  Meaning: if ping doesn't work, HTTP/HTTPS are less likely to work, too.

I think the ordering of these tests is what threw me.  😄

Reply:
> You might want to go from "smallest" to "biggest" test:

Yeah, good suggestion.  The ordering is another artifact of the way I developed the code, with maybe not the best end-result.  I added the HTTPS check first...then the PING, followed by the DNS name resolution, resolvers, and PING's last.  All provided some useful information in trying to resolve a very strange, intermittent issue I've seen at 5:00a every morning.  Though not every 5:00a run is a failure, most show symptoms of intermittent network issues...i.e, occasional HTTPS connection failures; some long PING (or timeouts), and unusually long DNS resolver PING's (or timeouts).  The approach for testing for "hard-failures" is more clear-cut than when dealing with intermittent ones I think...

> Ping is a fundamentally simpler network protocol than HTTP/HTTPS. I.e., it can verify basic networking connectivity (beyond DNS). Meaning: if ping doesn't work, HTTP/HTTPS are less likely to work, too.

Yeah, understand the differences with the various protocols, and your point is valid.  I'll look at ways to revise in a future release.

Reply:
Cool.

Do you want to merge this now?  Or do you want to keep working on this PR?  Either is fine -- the code is _correct_ as it is in this PR.  There's always an endless stream of improvements that can be made to any code.

Reply:
Hold off for now...I'm actually several versions ahead of this one now as it's been several weeks since I made the pull request, and I've addressed a few minor bugs since then.  I'll incorporate some of the discussion above and issue a new PR in the next day or two.  

On that note, is a `git rm ...` the best way to replace this PR with the new one, or is there another recommended workflow you suggest?  I expect this will continue to occur for future releases too (i.e., getting ahead of PR before it's been merged).

> There's always an endless stream of improvements that can be made to any code.

I can tell you write a lot of code...that's always the case with mine anyway!